src/NuGet.Core/NuGet.Protocol/README.md

```diff
@@ -1 +1,218 @@
-NuGet's implementation for interacting with feeds. Contains functionality for all feed types.
+# NuGet.Protocol
+
+NuGet.Protocol is a NuGet client SDK library that provides a set of APIs for interacting with NuGet feeds. It provides a way for developers to query NuGet feeds to discover packages and their dependencies, and also to download packages and their associated assets.
+
+## Getting started
+
+NuGet.Protocol can be installed from the NuGet Package Manager or using the dotnet CLI:
+
+```
+dotnet add package NuGet.Protocol
+```
+
+## Usage
+
+### List package versions
+
+Find all versions of Newtonsoft.Json using the [NuGet V3 Package Content API](https://learn.microsoft.com/en-us/nuget/api/package-base-address-resource#enumerate-package-versions):
+
+```c#
+ILogger logger = NullLogger.Instance;
+CancellationToken cancellationToken = CancellationToken.None;
+
+SourceCacheContext cache = new SourceCacheContext();
+SourceRepository repository = Repository.Factory.GetCoreV3("https://api.nuget.org/v3/index.json");
+FindPackageByIdResource resource = await repository.GetResourceAsync<FindPackageByIdResource>();
```

Comment:
I'm wondering if we should instead be describing the `resources` arch instead of just providing code samples.